### DISCUSSION TITLE
Notifications Library Suggestion - Display an icon when there are queued messages

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#69] [Feedback]
In my testing I sometimes do not dismiss notifications which are set to not `autoClose`...
This means sometimes there are queued messages waiting but I don't see.
There is no way for a user to know there is a queue of pending notifications.
For notifications which `autoClose` this is generally OK - but I still think there is a missing UI element here.

My idea is when you have the maximum number of notifications on display and there are still queued messages to be shown there should be some sort of UI flag.

Ideas:
- An icon with an ellipsis or pending message counter could appear beneath the most recent message (this icon would appear in place of the n+1 notification)
- The oldest message on display shakes briefly until closed? (could be annoying)
- Combination of both?

**Comment by rtivital** [comment]
This is not actually how package was designed to be used. The main purpose is to show notifications with autoClose or updated autoClose to attract user attention and give feedback on certain action. If you need persistent notifications – turning off autoClose is a bad option, it's much better to differentiate notifications in two types – flashy (@mantine/notifications) and persistent (separate state). 
The best place for persistent notifications in header with some badge icon, something like this in Material UI docs
<img width="52" alt="Screen Shot 2021-05-12 at 10 23 51 AM" src="https://user-images.githubusercontent.com/10353856/117935276-38de3c00-b30c-11eb-844b-514582c494a4.png">

**Reply by cstrat** [reply]
Yeah I might not have explained it property.
I wasn't wanting to use the notifications package as a persistent messaging system.

I was just pointing out that there is just no UI hint to show that there are further messages queued - and how many are queued.
It's probably an edge case. However even with autoClose, it could be possible someone is getting hit with more notifications than can be displayed and they might need to be aware of the growing message queue in the background.

Rather than just showing this (10 messages added to queue, only display 3)
![image](https://user-images.githubusercontent.com/4209076/118059459-ec5e3380-b3d3-11eb-9103-397b82a8a164.png)

Something simple like this:
![image](https://user-images.githubusercontent.com/4209076/118059833-c2594100-b3d4-11eb-9a35-59f8da3992a1.png)

There could be a flag in the `NotificationsProvider`, perhaps: `showQueueSummary`.
If that is true, whenever there are more messages it would just show a counter or an ellipse or something like that.

I did try to hack something together, but for some reason `notifications.queue` doesn't work as I'd expect, I thought I could read `notifications.queue.length` to get the length and perhaps do my own notification by updating the text of the last displayed notification but I can't even see the queue.

See here:
https://codesandbox.io/s/mantine-notification-hint-more-messages-hcymm?file=/src/App.js

Also - has this been brought up yet? I am seeing these warnings all over my console for my apps.
![image](https://user-images.githubusercontent.com/4209076/118060926-3d235b80-b3d7-11eb-9f28-fcdc2ecbc775.png)

**Reply by rtivital** [reply]
Missing queue is an issue, I forgot to add this, will do in #71 in next minor release (end of May), regarding `x more...` – I cannot think of a great solution and [it's usually not done in other libraries](https://fkhadra.github.io/react-toastify/limit-the-number-of-toast-displayed) so cannot promise anything.

**Reply by rtivital** [reply]
About jss issue, please create an issue and describe when these messages appear (OS, browser). I did not see this one yet, looks strange

**Reply by cstrat** [reply]
JSS issue is only on my Mac with Safari. Before I log an issue I will check to see if its already something known with JSS.

As for the `x more...` no worries, more important things to think about and you're probably right anyway.

**Reply by rtivital** [reply]
Thanks, sounds good
